### PR TITLE
tycho-its reproducer p2Repository.downloadVerifyNoDigestAlgo (#2710)

### PR DESCRIPTION
**Pull request by dhendriks**
My first merge request to contribute a tycho-its. Hope it works.

Addresses #2710

### CODE REVIEW

File: b/tycho-its/src/test/java/org/eclipse/tycho/test/p2Repository/DownloadVerifyNoDigestAlgo.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.apache.maven.it.Verifier;
+import org.eclipse.tycho.test.AbstractTychoIntegrationTest;
+import org.junit.Assert;
+
+public class DownloadVerifyNoDigestAlgo extends AbstractTychoIntegrationTest throws IOException {
```
Comment (laeubi) [inline b/tycho-its/src/test/java/org/eclipse/tycho/test/p2Repository/DownloadVerifyNoDigestAlgo.java:22]:
```suggestion
public class DownloadVerifyNoDigestAlgoTest extends AbstractTychoIntegrationTest throws IOException {

```

**Comment by dhendriks** [comment]
For some reason my new test is not picked up. Any idea why?

**Comment by dhendriks** [comment]
I think I found the problem. Let's see whether the next build works.

File: tycho-its/src/test/java/org/eclipse/tycho/test/p2Repository/DownloadVerifyNoDigestAlgoTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+public class DownloadVerifyNoDigestAlgoTest extends AbstractTychoIntegrationTest {
+    @Test
+    public void test() throws Exception {
+        Verifier verifier = getVerifier("p2Repository.downloadVerifyNoDigestAlgo", false);
+        verifier.setLogFileName("p2Repository.downloadVerifyNoDigestAlgo.log");
+        verifier.executeGoal("verify");
+        verifier.verifyErrorFreeLog();
+        String logContents = Files.readString(Paths.get("p2Repository.downloadVerifyNoDigestAlgo.log"));
+        assertFalse(logContents.contains("No digest algorithm is available to verify download"));
```
Comment (laeubi) [inline tycho-its/src/test/java/org/eclipse/tycho/test/p2Repository/DownloadVerifyNoDigestAlgoTest.java:31]: `AbstractTychoIntegrationTest` already contains some helper methods to check if log messages are (not) there.

**Comment by laeubi** [comment]
> The new integration test thus shows that Tycho currently gives the warnings that we don't want. What is the next step?

Please squash/rebase your change so it could be uses as a basis for a possible fix. Do you maybe want to give it a try reducing the message for some of the items? At least for feature/products I think it could be a similar fix as applied for plugins already. For m2e target items it should be a bit different and reusing the maven checksums.

**Comment by dhendriks** [comment]
> Please squash/rebase your change so it could be uses as a basis for a possible fix.

Done.

> Do you maybe want to give it a try reducing the message for some of the items?

It already took me way more time than expected to report the issue, make a minimal reproducer, and get that in the right form as a tycho-its (4+ hours). At the moment I don't have the time to actually go fixing this. Especially since I know absolutely nothing about the internals of Tycho, so it would be a huge effort to figure out where it is, what is going on, and how to address that. I was kind of hoping some Tycho developer could look at it?

**Pull request closed without merging**